### ISSUE TITLE
function to add the signature algorithm in the X509_CRL structure

### ISSUE THREAD

**Issue by hugill-8121** [openssl/openssl#18741]
Thanks for your time! I am not professional about this. In the `X509_CRL`  structure，there seems to be difficult to get the address of `crl.sig_alg`，while a similar function named `X509_get0_tbs_sigalg` exists in the series of functions about `X509`. the following is the reference code :

```
struct X509_crl_info_st {
    ASN1_INTEGER *version;      /* version: defaults to v1(0) so may be NULL */
    X509_ALGOR sig_alg;         /* signature algorithm */
    ......
};

struct X509_crl_st {
    X509_CRL_INFO crl;          /* signed CRL data */
    X509_ALGOR sig_alg;         /* CRL signature algorithm */
    .....
};
```
It leads to the fact that there is no way to set up the signature algorithm in this position very well when somebody can only take the approach of external signature because they don't own the private key . I can get the way to set the outer layer of the signature algorithm by call functions  `X509_CRL_get0_signature()` and `X509_ALGOR_set0()`  . 

I think it may be like the following:

```
const X509_ALGOR *X509_CRL_get0_tbs_sigalg(const X509_CRL *x)
{
    return &x->crl.sig_alg;
}
```
 By the way, I'm not sure there's any difference between the two position of signature algorithm ,either in the `X509` structure or in the `X509_CRL`  structure . They seem to be the same thing in my mind. A example is needed to me.

**Comment by davidben** [comment]
Mutating a const pointer seems ill-advised. Probably a better design would be to make a variant of X509_REQ_set1_signature_algo for the type. Since there's no reason to ever have the two mismatch, the function can just set both values concurrently.

**Comment by hugill-8121** [comment]
> Mutating a const pointer seems ill-advised. Probably a better design would be to make a variant of X509_REQ_set1_signature_algo for the type. Since there's no reason to ever have the two mismatch, the function can just set both values concurrently.

It seems that you are right. I got this idea about the following:
```
const X509_ALGOR *X509_get0_tbs_sigalg(const X509 *x)
{
    return &x->cert_info.signature;
}
```
I just wanted an easy way to set up the signature algorithm, rather than having to call the original function to get the signature and then override the signature value as I do now, which may be unexpected but is actually strange.

**Comment by nhorman** [comment]
marking as inactive, to be closed at the end of 3.4 dev barring further input
